addons/maptools/functions/fnc_canUseMapGPS.sqf

```diff
@@ -15,6 +15,14 @@
  * Public: No
  */
 
-visibleMap &&
-{alive ACE_player} &&
-{"ItemGPS" in (assignedItems ACE_player)}
+if (!visibleMap || {!alive ACE_player}) exitWith {false};
+
+private _playerTerminal = ACE_player getSlotItemName 612;
+[
+    "ItemGPS",
+    "B_UavTerminal",
+    "O_UavTerminal",
+    "I_UavTerminal",
+    "C_UavTerminal",
+    "I_E_UavTerminal"
```

Comment:
Should be item-agnostic, use item simulation or just check for presence of anything in GPS (or check for GPS side panel availability)

Reply:
Good point, since any vanilla terminal item gives GPS functionality. Not sure if there are any mod items for that slot that aren't GPSs though.

I don't think there is a function that can check for GPS Side Panel availability, only those that toggle it.

Reply:
https://community.bistudio.com/wiki/infoPanelComponentEnabled

Alternatively, https://community.bistudio.com/wiki/openGPS, `openGPS true` followed by `openGPS false`

Reply:
Oddly enough, `player infoPanelComponentEnabled ["left", "MinimapDisplay"];` always returns true, even when the terminal slot is empty.

Reply:
```sqf
(ace_player getSlotItemName 612) != ""
```
That should work for anything that goes in the GPS slot

Reply:
It does, I tested it.
I'm just not sure if there exists a mod which adds stuff to the GPS slot that doesn't have GPS functionality. But you're right, that's the slot's main purpose after all.

Reply:
The `openGPS` way seems like the most complete solution.

Reply:
https://github.com/johnb432/Tao-folding-map/blob/master/addons/main/functions/fnc_findGPS.sqf

Reply:
I probably misunderstood your message. A PFH that checks `visibleGPS` returns false when the GPS panel is available but not displayed.

Reply:
> I might have misunderstood your message, but a PFH that checks `visibleGPS` returns false when the GPS panel is available but not displayed.

I know of a mod that monitors the status of the GPS displays with `visibleGPS` every frame. It sends out a custom event, much like CBA, when something has changed. I was wondering if the quick opening and closing of the GPS could be detected by the mod. The best way to emulate the mod's behaviour is to add a PFH check and see what it returns.

Given your results suggest that it doesn't, I'm happy with this. In fact, I would like to adopt this code in my mod if you don't mind.

Reply:
Ah gotcha. I just tested that and it seems like a CBA PFH (interval 0) can't detect the `openGPS` code's opening and closing of the GPS.

> I would like to adopt this code in my mod if you don't mind.
For sure! Anything for less lag in Arma 🙏🏻